hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -148,10 +153,25 @@ private List<BlockedInfo> parseBlockList(final List<String> fileLines) {
     }
 
     @NonNull
-    private List<String> readPrivateKeyBlocklist(final @NonNull String fileName) {
-        final var inputStream = getClass().getClassLoader().getResourceAsStream(fileName);
-        final var reader = new BufferedReader(new InputStreamReader(inputStream));
-        return reader.lines().toList();
+    private List<String> readPrivateKeyBlocklist(
+            final @NonNull String defaultResource, final @NonNull String resourcePath) {
+        log.info("Bootstrapping blocklist from '{}'", resourcePath);
+        InputStream externalResourceStream = null;
+        try {
+            externalResourceStream = Files.newInputStream(Paths.get(resourcePath));
+        } catch (IOException e) {
+            log.warn("Could not read external properties file {}", resourcePath);
+        }
+
+        if (externalResourceStream != null) {
+            final var externalResourceReader = new BufferedReader(new InputStreamReader(externalResourceStream));
+            return externalResourceReader.lines().toList();
+        } else {
+            log.info("Bootstrapping blocklist from default resource '{}'", defaultResource);
+            final var inputStream = getClass().getClassLoader().getResourceAsStream(defaultResource);
```

Comment:
This will leak as well, the stream is never closed.
Also, it should be checked for null or missing resource.

Reply:
Updated to use try-with-resources.